sdk/tables/Azure.Data.Tables/src/Exceptions/TableBatchOperationFailedException.cs

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Runtime.Serialization;
+
+namespace Azure.Data.Tables
+{
+    /// <summary>
+    /// An exception thrown when a <see cref="TableTransactionalBatch"/> operation fails.
+    /// </summary>
+    public class TableBatchOperationFailedException : RequestFailedException
+    {
+        public TableBatchOperationFailedException(string message) : base(message)
+        {
+        }
+
+        public TableBatchOperationFailedException(string message, Exception innerException) : base(message, innerException)
+        {
+        }
+
+        public TableBatchOperationFailedException(int status, string message) : base(status, message)
+        {
+        }
+
+        public TableBatchOperationFailedException(int status, string message, Exception innerException) : base(status, message, innerException)
+        {
+        }
+
+        public TableBatchOperationFailedException(int status, string message, string errorCode, Exception innerException, ITableEntity entity) : base(status, message, errorCode, innerException)
+        {
+            TableEntity = entity;
+        }
+
+        protected TableBatchOperationFailedException(SerializationInfo info, StreamingContext context) : base(info, context)
+        {
+        }
+
+        /// <summary>
+        /// The <see cref="ITableEntity"/> related to the batch operation error.
+        /// </summary>
+        /// <value></value>
+        public ITableEntity TableEntity { get; }
```

Comment:
How would this be used?

Reply:
It's mostly a convenient way to get a reference to the entity that triggered the error in a batch. Scenarios I can imagine would be that they want to just inspect which row key was the problem, or casting it to the full type of their entity model to modify it and send in a new batch.

Reply:
This would be the first service-specific exception type we ship. So I'd like to make sure we really have the use-case for it.

Reply:
Can we use the Data property of RequestFailedException for this?

Reply:
I hadn't looked at that. Let me see if that would work.

Reply:
> Can we use the Data property of RequestFailedException for this?

I took a look - I had forgotten that `Data` is an `IDictionary`. I'm not sure what a reasonable key would be for the entity data. Have any suggestions?

Reply:
`"TableEntity"` ?

Reply:
I know it's a bit obscure. But so is the scenario :)

Reply:
I'm not sure I see the value of putting it in `Data` without a direct and obvious way to fetch it. Perhaps a reasonable alternative to the new exception type is to transform the exception message content to something like this:

before:
```json
{"odata.error":{"code":"EntityAlreadyExists","message":{"lang":"en-US","value":"3:The specified entity already exists.\nRequestId:db7ae526-e002-0004-6110-a72e1e000000\nTime:2020-10-20T18:42:21.6359859Z"}}}
```
after:
```
EntityAlreadyExists: The entity with RowKey="04" already exists. RequestId=db7ae526-e002-0004-6110-a72e1e000000, 
```

Then the developer would need to have stored the batch of entities themselves somehow to inspect. Not taking the precedent into consideration, I think the new type is the most user friendly.

Thoughts?

Reply:
I would be cool but wouldn't we have to support a lot of error message manipulation code if we go that route?

Reply:
True - that could be a pain to manage.

I like the new exception type best. Is there a concern with it other than it being rare?

Reply:
It's rare because most of the scenario that we intended to support with a new exception was very niche and not worth expanding the exception hierarchy. @KrzysztofCwalina is a large proponent of having as few exception types as possible.

Reply:
>I'm not sure I see the value of putting it in Data without a direct and obvious way to fetch it. 

Can we say in the message that Data["TableEntity"] contains the entity that failed?

Reply:
Discussed a bit offline. We'll add some additional context to the  message containing the RowKey of the entity causing the failure. If we get additional feedback that this is not sufficient, we'll reevaluate.